openmc/deplete/operator.py

```diff
@@ -251,7 +277,16 @@ def __call__(self, vec, power):
         time_openmc = time.time()
 
         # Extract results
-        op_result = self._unpack_tallies_and_normalize(power)
+        try:
+            op_result = self._unpack_tallies_and_normalize(power)
+        except ZeroDivisionError:
+            if comm.rank == 0:
+                sys.stderr.flush()
+                print(" No energy reported from openmc tallies. Do you have "
+                      "MT901 data?\n", file=sys.stderr, flush=True)
```

Comment:
Definitely agree. I can already imagine the frustration of starting a run, seeing it do through transport, only to find this out much later.

I think this would be a good fix in general. One way to go about this would be to have the `Tallies` do some checking when their scores and/or nuclides are set. As in
1) If no nuclides are set, are there *any* nuclides with data needed for this score?
2) Do *these specific nuclides* have data necessary for this score.

An extreme demonstration would be requesting recoverable fission energy from non-fissile isotopes. 

If we wanted to do this on the python side, we could expose the vector of all nuclides through `openmc.lib` and then be able to dig through `reaction_index_` to look for the 901 data.

Reply:
Exposing to the Python side seems a bit overkill. I think just having this check on the C++ side would be sufficient and doesn't require any extra bindings.